src/EventListener/InitializeSystemListener.php

```diff
@@ -388,17 +407,14 @@ private function handleRequestToken(Request $request = null)
         if ($_POST && null !== $request && !$this->tokenManager->isTokenValid($token)) {
             // Force a JavaScript redirect upon Ajax requests (IE requires absolute link)
             if ($request->isXmlHttpRequest()) {
-                header('HTTP/1.1 204 No Content');
-                header('X-Ajax-Location: ' . $this->router->generate('contao_backend'));
-            } else {
-                header('HTTP/1.1 400 Bad Request');
-                die_nicely(
-                    'be_referer',
-                    'Invalid request token. Please <a href="javascript:window.location.href=window.location.href">go back</a> and try again.'
-                );
+                throw AjaxRedirectResponseException::create($this->router->generate('contao_backend'));
             }
 
-            exit; // FIXME: throw a ResponseException instead
+            // FIXME: obsolete when using symfony security?
```

Comment:
I know it's not, but I wonder if that does matter? I mean the goal of our handler is to show pretty errors & hide sensitive information instead of rendering exceptions. To me it would totally make sense to render our pretty screen for appropriate Symfony exceptions too.